spec/src/main/asciidoc/core/spi.asciidoc

```diff
@@ -882,6 +885,29 @@ If any observer method of the `BeforeBeanDiscovery` event throws an exception, t
 
 If any `BeforeBeanDiscovery` method is called outside of the observer method invocation, an `IllegalStateException` is thrown.
 
+[[annotated_type_configurator]]
+
+===== `AnnotatedTypeConfigurator` interface
+
+CDI 2.0 introduced the `javax.enterprise.inject.spi.builder.AnnotatedTypeConfigurator` interface to help configuring a new `AnnotatedType` instance.
+
+`AnnotatedTypeConfigurator` must be initialized from a `java.lang.Class` or an existing `AnnotatedType` with one of its `read()` methods.
+
+With `AnnotatedTypeConfigurator` you can perform the following operations:
+
+* Add or remove annotation at type level with `addToType` and `removeFromType` methods.
+* Add or remove annotation on a field with `addToField` and `removeFromField` methods.
+* Add or remove annotation on a method with `addToMethod` and `removeFromMethod` methods.
+* Add or remove annotation on a method parameter with `addToMethodParameter` and `removeFromMethodParameter` methods.
+* Add or remove annotation on a constructor with `addToConstructor` and `removeFromConstructor` methods.
+* Add or remove annotation on a constructor parameter with `addToConstructorParameter` and `removeFromConstructorParameter` methods.
+* Remove an annotation from the type or any type member with `removeFromAll` method.
+* Override a field type with `overrideFieldType` methods.
+* Override a method parameter type with `overrideMethodParameterType` method.
+* Override a constructor parameter type with `overrideConstructorParameterType` method.
+* Change the class that the `AnnotatedType` will represent.
```

Comment:
NP. I have to find a better way to introduce those in the spec. Finally, I'm moving it in the Alternative metadata source section.